### PR TITLE
Cleanup of chef_handler examples + fix other resources

### PR DESCRIPTION
**Pull request by tas50**
More cleanup of chef_handler examples to make them potentially accurate

1) Remove references to the cookbook
2) Remove use of "supports' which is deprecated and will cause failures
3) Formatting fixes
4) Remove references to CloudKick

Also a lot of syncing of changes from chef/chef.

Signed-off-by: Tim Smith <tsmith@chef.io>

### CODE REVIEW

**Review by phiggins — APPROVED** [review]

File: content/resources/chef_handler/_index.md [hunk 1]
```diff
@@ -104,72 +91,148 @@ properties_list:
...
+    source 'chef/handler/json_file'
+    arguments path: '/var/chef/reports'
+    type exception: true
+    action :enable
+  end
+  ```
+
+  **Cookbook Versions (a custom handler)**
+
+  juliandunn created a custom report handler that logs all of the cookbooks and cookbook versions that were used during a Chef Infra Client run, and then reports after the run is complete. This handler requires the chef_handler resource (which is available from the chef_handler cookbook).
```
Comment (IanMadd) [inline content/resources/chef_handler/_index.md]:
```suggestion
  juliandunn created a custom report handler that logs all of the cookbooks and cookbook versions that were used during a Chef Infra Client run, and then reports after the run is complete. This handler requires the chef_handler resource, which is available from the chef_handler cookbook.
```

File: content/resources/chef_handler/_index.md [hunk 2]
```diff
@@ -104,72 +91,148 @@ properties_list:
...
+
+  [2013-11-26T03:11:06+00:00] INFO: Chef Infra Client Run complete in 0.300029878 seconds
+  [2013-11-26T03:11:06+00:00] INFO: Running report handlers
+  [2013-11-26T03:11:06+00:00] INFO: Cookbooks and versions run: ["cookbook_versions_handler 1.0.0"]
+  [2013-11-26T03:11:06+00:00] INFO: Report handlers complete
+  ```
+
+  **JsonFile Handler**
+
+  The json_file handler is available from the `chef_handler` cookbook and can be used with exceptions and reports. It serializes run status data to a JSON file. This handler may be enabled in one of the following ways.
```
Comment (IanMadd) [inline content/resources/chef_handler/_index.md]:
```suggestion
  The JsonFile handler is available from the `chef_handler` cookbook and can be used with exceptions and reports. It serializes run status data to a JSON file. This handler may be enabled in one of the following ways.
```

File: content/resources/chef_handler/_index.md [hunk 3]
```diff
@@ -104,72 +91,148 @@ properties_list:
...
+  chef_handler 'Chef::Handler::JsonFile' do
+    source 'chef/handler/json_file'
+    arguments path: '/var/chef/reports'
+    action :enable
+  end
+  ```
+
+  **ErrorReport Handler**
+
+  The error_report handler is built into Chef Infra Client and can be used for both exceptions and reports. It serializes error report data to a JSON file. This handler may be enabled in one of the following ways.
```
Comment (IanMadd) [inline content/resources/chef_handler/_index.md]:
```suggestion
  The ErrorReport Handler is built into Chef Infra Client and can be used for both exceptions and reports. It serializes error report data to a JSON file. This handler may be enabled in one of the following ways.
```

File: content/resources/openssl_ec_private_key/_index.md [hunk 4]
```diff
@@ -58,94 +49,71 @@ properties_list:
...
+  ```ruby
+  openssl_ec_private_key '/etc/ssl_files/eckey_prime256v1_des3.pem' do
+    key_curve 'prime256v1'
+    key_pass 'something'
+    action :create
+  end
+  ```
 
----
+  Generate a new ec private key with prime256v1 key curve and aes-128-cbc cipher
```
Comment (IanMadd) [inline content/resources/openssl_ec_private_key/_index.md]:
```suggestion
  Generate a new EC private key with prime256v1 key curve and aes-128-cbc cipher
```

File: content/resources/openssl_ec_public_key/_index.md [hunk 5]
```diff
@@ -8,127 +12,97 @@ menu:
...
-  Create a public key from a private key file\n\n  ``` ruby\n  openssl_ec_public_key\
-  \ '/etc/example/key.pub' do\n    private_key_path '/etc/example/key.pem'\n  end\n\
-  \  ```\n\n  Create a public key from a private key, without writing the private\n\
-  \  key to disk\n\n  You can provide the private key content as a string to the\n\
-  \  openssl_ec_public_key resource. In this example we just pass a\n  string, but\
-  \ this content could be loaded from an encrypted data bag or\n  other secure storage.\n\
-  \n  ``` ruby\n  openssl_ec_public_key '/etc/example/key.pub' do\n    private_key_content\
-  \ 'KEY_CONTENT_HERE_AS_A_STRING'\n  end\n  ```\n"
+examples: |
+  Generate new ec public key from a private key on disk
```
Comment (IanMadd) [inline content/resources/openssl_ec_public_key/_index.md]:
```suggestion
  Generate new EC public key from a private key on disk
```

File: content/resources/openssl_ec_public_key/_index.md [hunk 6]
```diff
@@ -8,127 +12,97 @@ menu:
...
----
+  ```ruby
+  openssl_ec_public_key '/etc/ssl_files/eckey_prime256v1_des3.pub' do
+    private_key_path '/etc/ssl_files/eckey_prime256v1_des3.pem'
+    private_key_pass 'something'
+    action :create
+  end
+  ```
+
+  Generate new ec public key by passing in a private key
```
Comment (IanMadd) [inline content/resources/openssl_ec_public_key/_index.md:95]:
```suggestion
  Generate new EC public key by passing in a private key
```

File: content/resources/openssl_rsa_private_key/_index.md [hunk 7]
```diff
@@ -8,144 +12,105 @@ menu:
...
   ruby_type: String
   required: false
   default_value: The resource block's name
-  new_in: null
   description_list:
-  - markdown: 'An optional property for specifying the path to write the file to if
+  - markdown: An optional property for specifying the path to write the file to if
+      it differs from the resource block's name.
+examples: |
+  Generate new 2048bit key with the default des3 cipher
```
Comment (IanMadd) [inline content/resources/openssl_rsa_private_key/_index.md:98]:
```suggestion
  **Generate new 2048bit key with the default des3 cipher**
```

File: content/resources/openssl_rsa_private_key/_index.md [hunk 8]
```diff
@@ -8,144 +12,105 @@ menu:
...
-  \  end\n  ```\n"
+  ```ruby
+  openssl_rsa_private_key '/etc/ssl_files/rsakey_des3.pem' do
+    key_length 2048
+    action :create
+  end
+  ```
 
----
+  Generate new 1024bit key with the aes-128-cbc cipher
```
Comment (IanMadd) [inline content/resources/openssl_rsa_private_key/_index.md:107]:
```suggestion
  **Generate new 1024bit key with the aes-128-cbc cipher**
```

File: content/resources/openssl_rsa_public_key/_index.md [hunk 9]
```diff
@@ -47,87 +44,91 @@ properties_list:
...
-  Create a public key from a private key file\n\n  ``` ruby\n  openssl_rsa_public_key\
-  \ '/etc/example/key.pub' do\n    private_key_path '/etc/example/key.pem'\n  end\n\
-  \  ```\n\n  Create a public key from a private key, without writing the private\n\
-  \  key to disk\n\n  You can provide the private key content as a string to the\n\
-  \  openssl_rsa_public_key resource. In this example we just pass a\n  string, but\
-  \ this content could be loaded from an encrypted data bag or\n  other secure storage.\n\
-  \n  ``` ruby\n  openssl_rsa_public_key '/etc/example/key.pub' do\n    private_key_content\
-  \ 'KEY_CONTENT_HERE_AS_A_STRING'\n  end\n  ```\n"
+examples: |
+  Generate new public key from a private key on disk
```
Comment (IanMadd) [inline content/resources/openssl_rsa_public_key/_index.md:85]:
```suggestion
  **Generate new public key from a private key on disk**
```

File: content/resources/openssl_rsa_public_key/_index.md [hunk 10]
```diff
@@ -47,87 +44,91 @@ properties_list:
...
----
+  ```ruby
+  openssl_rsa_public_key '/etc/ssl_files/rsakey_des3.pub' do
+    private_key_path '/etc/ssl_files/rsakey_des3.pem'
+    private_key_pass 'something'
+    action :create
+  end
+  ```
+
+  Generate new public key by passing in a private key
```
Comment (IanMadd) [inline content/resources/openssl_rsa_public_key/_index.md:95]:
```suggestion
  **Generate new public key by passing in a private key**
```

File: content/resources/openssl_x509_certificate/_index.md [hunk 11]
```diff
@@ -70,248 +69,184 @@ properties_list:
...
-      a key from this location. If no key file is found, the resource will
-
-      generate a new key file at this location. If the key_file property
-
-      is not specified, the resource will generate a key file in the same
-
-      directory as the generated certificate, with the same name as the
-
-      generated certificate.'
+  - markdown: The path to a certificate key file on the filesystem. If the key_file
```
Comment (IanMadd) [inline content/resources/openssl_x509_certificate/_index.md:146]:
```suggestion
  - markdown: The path to a certificate key file on the filesystem. If the `key_file`
```

File: content/resources/openssl_x509_certificate/_index.md [hunk 12]
```diff
@@ -70,248 +69,184 @@ properties_list:
...
+      property is specified, the resource will attempt to source a key from this location.
+      If no key file is found, the resource will generate a new key file at this location.
+      If the key_file property is not specified, the resource will generate a key
```
Comment (IanMadd) [inline content/resources/openssl_x509_certificate/_index.md:149]:
```suggestion
      If the `key_file` property is not specified, the resource will generate a key
```

File: content/resources/openssl_x509_certificate/_index.md [hunk 13]
```diff
@@ -70,248 +69,184 @@ properties_list:
...
   ruby_type: Integer
   required: false
   default_value: '2048'
-  new_in: null
+  allowed_values: 1024, 2048, 4096, 8192
   description_list:
-  - markdown: 'The desired bit length of the generated key (if key_type is equal
-
-      to ''rsa''). Available options are `1024`, `2048`, `4096`, and `8192`.'
+  - markdown: The desired bit length of the generated key (if key_type is equal to
```
Comment (IanMadd) [inline content/resources/openssl_x509_certificate/_index.md:158]:
```suggestion
  - markdown: The desired bit length of the generated key (if `key_type` is equal to
```

File: content/resources/openssl_x509_certificate/_index.md [hunk 14]
```diff
@@ -70,248 +69,184 @@ properties_list:
...
   required: false
-  default_value: '"prime256v1"'
-  new_in: null
+  default_value: prime256v1
+  allowed_values: '"prime256v1", "secp384r1", "secp521r1"'
   description_list:
-  - markdown: 'The desired curve of the generated key (if key_type is equal to
-
-      ''ec''). Run `openssl ecparam -list_curves` to see available options.'
+  - markdown: The desired curve of the generated key (if key_type is equal to 'ec').
```
Comment (IanMadd) [inline content/resources/openssl_x509_certificate/_index.md:140]:
```suggestion
  - markdown: The desired curve of the generated key (if `key_type` is equal to 'ec').
```

File: content/resources/openssl_x509_certificate/_index.md [hunk 15]
```diff
@@ -70,248 +69,184 @@ properties_list:
...
-  \     'keyUsage' => {\n        'values' => %w(\n          keyEncipherment\n    \
-  \      digitalSignature),\n        'critical' => true,\n      },\n      'extendedKeyUsage'\
-  \ => {\n        'values' => %w(serverAuth),\n        'critical' => false,\n    \
-  \  }\n    )\n    subject_alt_name ['IP:127.0.0.1', 'DNS:localhost.localdomain']\n\
-  \  end\n  ```\n"
-
+  - markdown: Array of Subject Alternative Name entries, in format `DNS:example.com`
+      or `IP:1.2.3.4`.
+examples: |
+  Create a simple self-signed certificate file
```
Comment (IanMadd) [inline content/resources/openssl_x509_certificate/_index.md:218]:
```suggestion
  **Create a simple self-signed certificate file**
```

File: content/resources/openssl_x509_certificate/_index.md [hunk 16]
```diff
@@ -70,248 +69,184 @@ properties_list:
...
+  ```ruby
+  openssl_x509_certificate '/etc/httpd/ssl/mycert.pem' do
+    common_name 'www.f00bar.com'
+    org 'Foo Bar'
+    org_unit 'Lab'
+    country 'US'
+  end
+  ```
+
+  Create a certificate using additional options
```
Comment (IanMadd) [inline content/resources/openssl_x509_certificate/_index.md:229]:
```suggestion
  **Create a certificate using additional options**
```

File: content/resources/openssl_x509_request/_index.md [hunk 17]
```diff
@@ -54,165 +55,131 @@ properties_list:
...
   required: false
-  default_value: '"prime256v1"'
-  new_in: null
+  default_value: prime256v1
+  allowed_values: '"prime256v1", "secp384r1", "secp521r1"'
   description_list:
-  - markdown: 'The desired curve of the generated key (if key_type is equal to
-
-      ''ec''). Run `openssl ecparam -list_curves` to see available options.'
+  - markdown: The desired curve of the generated key (if key_type is equal to `ec`).
```
Comment (IanMadd) [inline content/resources/openssl_x509_request/_index.md:86]:
```suggestion
  - markdown: The desired curve of the generated key (if `key_type` is equal to `ec`).
```

File: content/resources/openssl_x509_request/_index.md [hunk 18]
```diff
@@ -54,165 +55,131 @@ properties_list:
...
-      is not specified, the resource will generate a key file in the same
-
-      directory as the generated certificate, with the same name as the
-
-      generated certificate.'
+  - markdown: The path to a certificate key file on the filesystem. If the `key_file`
+      property is specified, the resource will attempt to source a key from this location.
+      If no key file is found, the resource will generate a new key file at this location.
+      If the `key_file` property is not specified, the resource will generate a key
+      file in the same directory as the generated certificate, with the same name
```
Comment (IanMadd) [inline content/resources/openssl_x509_request/_index.md:96]:
```suggestion
      file in the same directory as the generated certificate with the same name
```

File: content/resources/openssl_x509_request/_index.md [hunk 19]
```diff
@@ -54,165 +55,131 @@ properties_list:
...
-  \  ```\n\n  Create a new certificate request file from an existing ec key\n\n  ```\
-  \ ruby\n  openssl_x509_request '/etc/ssl_files/my_ec_request2.csr' do\n     common_name\
-  \ 'myecrequest2.example.com'\n     org 'Test Kitchen Example'\n     org_unit 'Kitchens'\n\
-  \     country 'UK'\n     key_file '/etc/ssl_files/my_ec_request.key'\n  end\n  ```\n\
-  \n  Create both a new rsa key and certificate request file\n\n  ``` ruby\n  openssl_x509_request\
-  \ '/etc/ssl_files/my_rsa_request.csr' do\n     common_name 'myrsarequest.example.com'\n\
-  \     org 'Test Kitchen Example'\n     org_unit 'Kitchens'\n     country 'UK'\n\
-  \     key_type 'rsa'\n  end\n  ```\n"
+examples: |
+  **Generate new ec key and csr file**
```
Comment (IanMadd) [inline content/resources/openssl_x509_request/_index.md]:
```suggestion
  **Generate new EC key and CSR file**
```

File: content/resources/openssl_x509_request/_index.md [hunk 20]
```diff
@@ -54,165 +55,131 @@ properties_list:
...
+  ```ruby
+  openssl_x509_request '/etc/ssl_files/my_ec_request.csr' do
+    common_name 'myecrequest.example.com'
+    org 'Test Kitchen Example'
+    org_unit 'Kitchens'
+    country 'UK'
+  end
+  ```
+
+  **Generate a new csr file from an existing ec key**
```
Comment (IanMadd) [inline content/resources/openssl_x509_request/_index.md]:
```suggestion
  **Generate a new CSR file from an existing EC key**
```

File: content/resources/openssl_x509_request/_index.md [hunk 21]
```diff
@@ -54,165 +55,131 @@ properties_list:
...
+  openssl_x509_request '/etc/ssl_files/my_ec_request2.csr' do
+    common_name 'myecrequest2.example.com'
+    org 'Test Kitchen Example'
+    org_unit 'Kitchens'
+    country 'UK'
+    key_file '/etc/ssl_files/my_ec_request.key'
+  end
+  ```
+
+  **Generate new rsa key and csr file**
```
Comment (IanMadd) [inline content/resources/openssl_x509_request/_index.md]:
```suggestion
  **Generate new RSA key and CSR file**
```

File: content/resources/powershell_package_source/_index.md [hunk 22]
```diff
@@ -53,83 +47,45 @@ properties_list:
...
 - property: publish_location
   ruby_type: String
   required: false
-  default_value: null
-  new_in: null
   description_list:
-  - markdown: 'The url where modules will be published to for this source. Only
-
-      valid if the provider is ''PowerShellGet''.'
+  - markdown: The url where modules will be published to for this source. Only valid
```
Comment (IanMadd) [inline content/resources/powershell_package_source/_index.md]:
```suggestion
  - markdown: The URL where modules will be published to for this source. Only valid
```

File: content/resources/powershell_package_source/_index.md [hunk 23]
```diff
@@ -53,83 +47,45 @@ properties_list:
...
 - property: script_source_location
   ruby_type: String
   required: false
-  default_value: null
-  new_in: null
   description_list:
-  - markdown: 'The url where scripts are located for this source. Only valid if the
-
-      provider is ''PowerShellGet''.'
+  - markdown: The url where scripts are located for this source. Only valid if the
```
Comment (IanMadd) [inline content/resources/powershell_package_source/_index.md]:
```suggestion
  - markdown: The URL where scripts are located for this source. Only valid if the
```

**Review by IanMadd — COMMENTED** [review]
A bunch of minor corrections

File: content/resources/chef_handler/_index.md (hunk 1 above)
Comment (tas50) [reply content/resources/chef_handler/_index.md]: Fixing this by removing this line instead

**Review by IanMadd — APPROVED** [review]

**Pull request merged**